packages/ERTP/src/typeGuards.js

```diff
@@ -129,3 +129,95 @@ export const DisplayInfoShape = M.partial(
     // properties beyond those in the `base` record.
   }),
 );
+
+// //////////////////////// Interfaces /////////////////////////////////////////
+
+export const BrandShape = M.remotable('Brand');
+export const IssuerShape = M.remotable('Issuer');
+export const PaymentShape = M.remotable('Payment');
+export const PurseShape = M.remotable('Purse');
+export const DepositFacetShape = M.remotable('DepositFacet');
+const NotifierShape = M.remotable('Notifier');
+export const MintShape = M.remotable('Mint');
+
+export const BrandI = M.interface('Brand', {
+  isMyIssuer: M.callWhen(M.await(IssuerShape)).returns(M.boolean()),
+  getAllegedName: M.call().returns(M.string()),
+  getDisplayInfo: M.call().returns(DisplayInfoShape),
+  getAmountSchema: M.call().returns(M.pattern()),
+});
+
+/**
+ * @param {Pattern} [brand]
+ * @param {Pattern} [assetKind]
+ * @param {Pattern} [amountSchema]
+ */
+export const makeIssuerInterfaces = (
+  brand = BrandShape,
+  assetKind = AssetValueShape,
+  amountSchema = AmountShape,
+) => {
+  const IssuerI = M.interface('Issuer', {
+    getBrand: M.call().returns(brand),
+    getAllegedName: M.call().returns(M.string()),
+    getAssetKind: M.call().returns(assetKind),
+    getDisplayInfo: M.call().returns(DisplayInfoShape),
+    makeEmptyPurse: M.call().returns(PurseShape),
+
+    isLive: M.callWhen(M.await(PaymentShape)).returns(M.boolean()),
+    getAmountOf: M.callWhen(M.await(PaymentShape)).returns(amountSchema),
+    burn: M.callWhen(M.await(PaymentShape))
+      .optional(M.pattern())
+      .returns(amountSchema),
+    claim: M.callWhen(M.await(PaymentShape))
+      .optional(M.pattern())
+      .returns(PaymentShape),
+    combine: M.call(M.arrayOf(M.eref(PaymentShape)))
+      .optional(amountSchema)
+      .returns(M.eref(PaymentShape)),
+    split: M.callWhen(M.await(PaymentShape), amountSchema).returns(
+      M.arrayOf(PaymentShape),
+    ),
+    splitMany: M.callWhen(
+      M.await(PaymentShape),
+      M.arrayOf(amountSchema),
+    ).returns(M.arrayOf(PaymentShape)),
+  });
+
+  const MintI = M.interface('Mint', {
+    getIssuer: M.call().returns(IssuerShape),
+    mintPayment: M.call(amountSchema).returns(PaymentShape),
+  });
+
+  const PaymentI = M.interface('Payment', {
+    getAllegedBrand: M.call().returns(brand),
+  });
+
+  const PurseI = M.interface('Purse', {
+    getAllegedBrand: M.call().returns(brand),
+    getCurrentAmount: M.call().returns(amountSchema),
+    getCurrentAmountNotifier: M.call().returns(NotifierShape),
+    deposit: M.call(PaymentShape).optional(M.pattern()).returns(amountSchema),
```

Comment:
This deserves a comment saying that it's intentional that it doesn't use `callWhen`.

Reply:
Done